play/src/main/scala/com/gu/pandahmac/HmacSecrets.scala

```diff
@@ -0,0 +1,28 @@
+package com.gu.pandahmac
+
+import com.gu.hmac.ValidateHMACHeader
+
+import java.net.URI
+
+trait HMACSecrets extends ValidateHMACHeader {
+  @deprecated("In an upcoming release consumers will expected to provide a list of secrets to `secretKeys` to allow for secret rotation.")
+  def secret: String = ""
+  def secretKeys: List[String] = List.empty
+
+  protected def gatherSecrets = {
+    val gatheredSecrets = (secret, secretKeys) match {
+      case ("", emptyList) if emptyList.isEmpty => List.empty
```

Comment:
```suggestion
      case ("", List()) => List.empty
```
or 
```suggestion
      case ("", Nil) => List.empty
```

Reply:
I think a case statement might not be the best control structure here, will refactor and prefer `secretKeys` if both it and `secret` are provided.